src/nodes/Geometry3D/IndexedFaceSet.js

```diff
@@ -932,111 +943,18 @@ x3dom.registerNodeType(
                                             this._mesh._texCoords[0].push(texCoords[t2].z);
                                         }
                                     }
-
```

Comment:
not sure. The switch logic is harder to follow, so I am not confident this can be removed. Note the p1=p2;t1=t2 setup. Since there is a lot of code repetition, it would make sense to reuse code in functions. Why not just leave it alone ?